storage/tokudb/tokudb_sysvars.cc

```diff
@@ -918,7 +923,91 @@ static MYSQL_THDVAR_BOOL(
     true);
 #endif
 
+static void dir_cmd_update_last_error_str(THD* thd,
+                                          struct st_mysql_sys_var* var,
+                                          void* var_ptr,
+                                          const void* save);
+
+static int dir_cmd_check(THD* thd, struct st_mysql_sys_var* var,
+                         void* save, struct st_mysql_value* value) ;
+
+static MYSQL_THDVAR_ULONG(dir_cmd_last_error,
+    PLUGIN_VAR_THDLOCAL,
+    "error from the last dir command. 0 is success",
+    NULL, NULL, 0, 0, ~0ULL, 1);
+
+static MYSQL_THDVAR_STR(dir_cmd_last_error_string,
+    PLUGIN_VAR_THDLOCAL,
+    "error string from the last dir command",
+    NULL, dir_cmd_update_last_error_str, NULL);
+
+static MYSQL_THDVAR_STR(dir_cmd,
+    PLUGIN_VAR_THDLOCAL + PLUGIN_VAR_MEMALLOC,
+    "name of the directory where the backup is stored",
+    dir_cmd_check, NULL, NULL);
+
+static void dir_cmd_set_error(THD *thd,
+                              int error,
+                              const char *error_fmt,
+                              ...) {
+    THDVAR(thd, dir_cmd_last_error) = error;
+    char *old_error_string = THDVAR(thd, dir_cmd_last_error_string);
+    if (error_fmt) {
+        va_list varargs;
+        va_start(varargs, error_fmt);
+        char   buff[error_buffer_max_size];
+
+        vsnprintf(buff, sizeof(buff), error_fmt, varargs);
+
+        THDVAR(thd, dir_cmd_last_error_string) =
```

Comment:
Sure, I will recheck this moment. But it looks like the issue is the same as described here https://bugs.launchpad.net/percona-server/+bug/1671082/comments/1 for TokuDB Hotbackup.